pages/posts/page/[page_number].tsx

```diff
@@ -0,0 +1,87 @@
+import fs from 'node:fs';
+import type { GetStaticPaths, GetStaticProps } from 'next';
+import type { DualPost } from '../../../lib/types';
+import PostPreview from '../../../components/PostPreview';
+import { getAllPosts } from '../../../lib/getAllPosts';
+import { podcastXml } from '../../../lib/podcast';
+import PageWrapper from '../../../components/PageWrapper';
+import Paginator from '../../../components/Paginator';
+import { paginate } from '../../../lib/helpers';
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const allPosts = await getAllPosts();
+  const whichPage = Number(context.params?.page_number);
+  const thisPagePosts = paginate(
+    allPosts.sort(
+      (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+    ),
+    whichPage,
+    8,
+  );
+  const enPosts = allPosts.map((dual) => dual.en);
+  const esPosts = allPosts.map((dual) => dual.es);
+  fs.writeFileSync(`./public/podcast.en.rss`, podcastXml(enPosts));
+  fs.writeFileSync(`./public/podcast.es.rss`, podcastXml(esPosts));
+
+  return {
+    props: {
+      posts: thisPagePosts,
+      pageCount: Math.ceil(allPosts.length / 8),
+      pageNum: whichPage,
+    },
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
```

Comment:
come to think of it, this might be a good place to move the podcast generation... i think this will only get called once?